monero/caip10.md

```diff
@@ -0,0 +1,76 @@
+---
+namespace-identifier: monero-caip10
+title: Monero - Addresses
+author: silverpill (@silverpill)
+discussions-to: https://github.com/ChainAgnostic/namespaces/issues/41
+status: Draft
+type: Informational
+created: 2023-09-01
+requires: ["CAIP-10"]
+---
+
+# CAIP-10
+
+*For context, see the [CAIP-10][] specification.*
+CAIP-10 defines an pattern for identifying addresses in a cryptographic system as heirarchical URIs.
```

Comment:
To be totally honest, CAIP-10 is final but I feel like the semantics of "account" aren't really explicit enough; if I can get some UTXO people to help me write it I might try writing an `Informational` CAIP to caveat and nuance this mapping a little, which is why I'm being so stubborn here about what does and doesn't work in the equivalence!

CAIP-10 doesn't really say if by "account" it means:
1. a "unit of accounting" (as in balance, whether public or requiring a key to see),
1. "distinct unit of account" (as in, a balance independent from any other),
1. "address" (string expresses a stable, addressable identity), AND
1. "actor" (as in payer/payee in a ledger-like transaction system).

In EVM and similar "account-based" networks, each wallet's address or CAIP-10 represents exactly one of each.  But in a system where there are not 1:1 relationships between the four, it's not self-evident which interpretation best serves interoperability across systems.

I have been going on the assumption that # 3 above is the most important, because the other three options make far less sense reading them against the [## Semantics section of btc/caip10.md](https://namespaces.chainagnostic.org/bip122/caip10#rationale), which says P2SH addresses are "close enough" to the EVM account model and implies by omission that earlier P2PKH addresses are not.  Since a given wallet/privatekey can generate multiple P2SH addresses and these payscripts can impose conditions, timelocks, etc, none of the other 3 seems valid.  To further undermine my confidence here, though, the example given by that profile is not, in fact, a P2SH address, but a P2PKH one (it starts with 1 instead of 3 in base58 form).  Later address types (taproot, segwit) aren't addressed at all!

I guess what I'm saying is that I'm over-editing your first line because I'm trying to make your summary of CAIP-10 more explicit than CAIP-10 is 😅

Reply:
I think of accounts as boxes that hold certain amounts of money. In Ethereum-like systems the balance is continuous: you can subtract or add any amount. In UTXO-based systems the box actually contains discrete "notes": you can take one note out and put a smaller one back, for example. It's called "address" but it nevertheless works like an account.

Some Bitcoin and Monero software introduce another level of abstraction ("wallets" or "accounts") in attempt to simplify address management, but in my opinion it's outside of CAIP-10 scope.

So in most (if not all) cases 1 and 3 will be the same thing (I don't quite understand the difference between 1 and 2). 4 is largely depends on how account is used. Sometimes it represents a unique identity. Sometimes several accounts are linked the one person.

>which says P2SH addresses are "close enough" to the EVM account model and implies by omission that earlier P2PKH addresses are not.

IIRC both P2SH and P2PKH can have balances and therefore can be viewed as CAIP-10 accounts. Same for segwit, taproot and others.

Reply:
I think the trick here is that wallets aggregate accounts. for example, wallet A has accounts X, Y, and Z in them. each have 10btc in them. A can send someone 30btc taking notes out of all three-- that's what I meant by "independent" in #2. they can *operate* independently and each have a different balance, yet a single key/actor controls all three so that "independence" is only contingent on the information assymetry between A and its counterparties who don't realize those independent accounts are really one actor and can function as one account at will 😄

anyways thanks for thinking thru this with me, I think there's no change worth making here I just wanted to explore the mental model a bit!